internal/lib/lib.go

```diff
@@ -671,20 +671,59 @@ func SetClusterLabelChecksum() {
 		clusterLabelChecksum = utils.Hash(clusterKey + clusterValue)
 	}
 }
-
 func GetClusterLabelChecksum() uint32 {
 	return clusterLabelChecksum
 }
+func GetMarkersChecksum(markers utils.AviObjectMarkers) uint32 {
+	vals := reflect.ValueOf(markers)
+	var j int
+	var cksum uint32
+	var combinedString string
+	numMarkerFields := vals.NumField()
+	markersStr := make([]string, numMarkerFields)
+	for i := 0; i < numMarkerFields; i++ {
+		if vals.Field(i).Interface() != "" {
+			value := vals.Field(i).Interface().(string)
+			markersStr[j] = value
+			j = j + 1
+		}
+	}
+	sort.Strings(markersStr)
+	for _, ele := range markersStr {
+		combinedString += ele
+	}
+	if len(combinedString) != 0 {
```

Comment:
why are you doing a length check on a string variable?

Reply:
@sudswasavi : GetMarkersChecksum function is common for shared objects and individual objects. For Shared objects "combinedString" will be empty length. This empty length string produces non zero checksum value. This will result in PUT call for existing shared objects even though there is no change in markers to those objects. (On previous release, these objects are marked with only Clustername label.)
To avoid un-necessary PUT call, this check is there.